Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -464,11 +477,21 @@ void TextureCacheBase::BindTextures()
 
 void TextureCacheBase::UnbindTextures()
 {
-  bound_textures.fill(nullptr);
+  for (size_t i = 0; i < bound_textures.size(); ++i)
```

Comment:
In `TextureCacheBase::Invalidate()` i'm just going to call `InvalidateAllBindPoints();` before `UnbindTextures();`. Thanks for pointing that out, it wouldn't have deleted all textures.

I don't understand your isssue with the call to `UnbindTextures();` in `VertexManagerBase::Flush()`. It always goes through `UnbindTextures();`, where it only unbinds all the invalidated binds, then it loads all used textures, and then (re)binds all textures that are not invalidated. The only thing i see that is a bit weird is that it would rebind textures that are not invalided, but also are not used. Not sure if skipping that would optimize things, or make things worse, if those textures are used again later, with no invalidation in between.